libafl/src/executors/inprocess.rs

```diff
@@ -570,16 +581,57 @@ pub fn run_observers_and_save_state<E, EM, OF, Z>(
 ) where
     E: HasObservers,
     EM: EventFirer<State = E::State> + EventRestarter<State = E::State>,
+    CF: Feedback<E::State>,
     OF: Feedback<E::State>,
-    E::State: HasSolutions + HasClientPerfMonitor + HasCorpus,
-    Z: HasObjective<Objective = OF, State = E::State>,
+    E::State: HasSolutions + HasClientPerfMonitor + HasCorpus + HasExecutions,
+    Z: HasObjective<Objective = OF, State = E::State>
+        + HasFeedback<Feedback = CF, State = E::State>
+        + HasScheduler,
 {
     let observers = executor.observers_mut();
 
     observers
         .post_exec_all(state, input, &exitkind)
         .expect("Observers post_exec_all failed");
 
+    let interesting_for_corpus = fuzzer
+        .feedback_mut()
+        .is_interesting(state, event_mgr, input, observers, &exitkind)
+        .expect("In run_observers_and_save_state feedback failure.");
+
+    if interesting_for_corpus {
+        let mut new_testcase = Testcase::with_executions(input.clone(), *state.executions());
+        new_testcase.add_metadata(exitkind);
+        new_testcase.set_parent_id_optional(*state.corpus().current());
+        fuzzer
+            .feedback_mut()
+            .append_metadata(state, observers, &mut new_testcase)
+            .expect("Failed adding metadata");
+        let idx = state
+            .corpus_mut()
+            .add(new_testcase)
+            .expect("In run_observers_and_save_state corpus failure.");
+        fuzzer
+            .scheduler_mut()
+            .on_add(state, idx)
+            .expect("Could not add to scheduler in run_observers_and_save_state.");
+        event_mgr
+            .fire(
+                state,
+                Event::NewTestcase {
```

Comment:
The main issue is that this could make your fuzzer _sloooow_, but it's the right thing to do afaik, anyway :)